src/features/device-state/utils.ts

```diff
@@ -33,10 +33,14 @@ export const serviceInstallFromImage = (
 	}
 
 	let id: number;
-	if (typeof image.is_a_build_of__service === 'object') {
-		id = image.is_a_build_of__service.__id;
+	if (Array.isArray(image.is_a_build_of__service)) {
```

Comment:
I think this is invalid without other changes to functionality in the code to `$expand` `is_a_build_of__service`? It also makes this a major change as this is part of the exported api surface of this module (and it gets used by cloud api in order to support the legacy state endpoint v1)

Reply:
revereted